addons/account_edi_ubl_cii/models/account_edi_common.py

```diff
@@ -680,68 +682,71 @@ def _retrieve_fixed_tax(self, invoice_line, fixed_tax_vals):
                     return tax
         return self.env['account.tax']
 
-    def _retrieve_taxes(self, invoice_line, line_values):
+    def _retrieve_taxes(self, record, line_values, tax_type):
         """
-        Retrieve the taxes on the invoice line at import.
+        Retrieve the taxes on the document line at import.
 
         In a UBL/CII xml, the Odoo "price_include" concept does not exist. Hence, first look for a price_include=False,
         if it is unsuccessful, look for a price_include=True.
         """
         # Taxes: all amounts are tax excluded, so first try to fetch price_include=False taxes,
         # if no results, try to fetch the price_include=True taxes. If results, need to adapt the price_unit.
         logs = []
-        line_values['tax_ids'] = []
+        taxes = []
         for tax_node in line_values.pop('tax_nodes'):
             amount = float(tax_node.text)
             domain = [
-                *self.env['account.journal']._check_company_domain(invoice_line.company_id),
+                *self.env['account.journal']._check_company_domain(record.company_id),
                 ('amount_type', '=', 'percent'),
-                ('type_tax_use', '=', invoice_line.move_id.journal_id.type),
+                ('type_tax_use', '=', tax_type),
                 ('amount', '=', amount),
             ]
             tax = self.env['account.tax']
-            if hasattr(invoice_line, '_predict_specific_tax'):
-                # company check is already done in the prediction query
-                predicted_tax_id = invoice_line \
-                    ._predict_specific_tax('percent', amount, invoice_line.move_id.journal_id.type)
-                tax = self.env['account.tax'].browse(predicted_tax_id)
+            if hasattr(record, '_get_specific_tax'):
+                tax = record._get_specific_tax(line_values['name'], 'percent', amount, tax_type)
```

Comment:
can't we simply do
`if hasattr(record, '_predict_specific_tax'):`
and keep the existing lines as is ? we could then remove `_get_specific_tax`

Reply:
Also see my other comment in the enterprise part: I'm not sure `_predict_specific_tax` should be api.model

Reply:
To be able to avoid creating lines to predict specific tax we are making `_predict_specific_tax` model method